doc/examples/move_group_capabilities/move_group_capabilities.rst

```diff
@@ -0,0 +1,85 @@
+
+
+Getting Started
+---------------
+If you haven't already done so, make sure you've completed the steps in :doc:`Getting Started </doc/tutorials/getting_started/getting_started>`.
+
+You should also have gone through the steps in :doc:`Visualization with MoveIt RViz Plugin </doc/tutorials/quickstart_in_rviz/quickstart_in_rviz_tutorial>`
+
+Creating a MoveGroup Capability
+-------------------------------
+`This Page <http://wiki.ros.org/pluginlib>`_ gives a detailed explanation of how to add plugins in ROS in general. The two necessary elements are base and plugin classes. The plugin class inherits from the base class and overrides its virtual functions. The main library used for this purpose is pluginlib.
+
+In this section, we will show how to add a new MoveGroup capability. For demonstration purposes, we will create a capability which provide a service to query the collision objects in MoveGroup's planning scene. The final source files designed in this tutorial are available :codedir:`here <move_group_capabilities/src>`.
+
+First we start by creating a class GetCollisionObjects by inheriting from move_group::MoveGroupCapability, to do so we create a file named ``get_collision_objects_service_capability.cpp`` in src folder. the minimal requirement is to pass the plugin name to the base class constructor ``MoveGroupCapability``, override the ``initialize`` function, and register the class as a plugin by using ``PLUGINLIB_EXPORT_CLASS``.
+
+.. tutorial-formatter:: ./src/get_collision_objects_service_capability.cpp
+
+Exporting the capability
+^^^^^^^^^^^^^^^^^^^^^^^^
+To export the plugin we create a plugin description xml file ``plugin_description.xml``
+
+.. code-block:: xml
+
+  <library path="move_group_tool_changing_capability">
+    <class name="move_group/ToolChangingCapability" type="move_group::ToolChangingCapability" base_class_type="move_group::MoveGroupCapability">
+        <description>
+            MoveGroup capability to get a list of collision objects in the planning scene
+        </description>
+    </class>
+  </library>
+
+The ``plugin_description.xml`` need to be added to the ament index to be found at runtime, in ``CMakeLists.txt`` add
+
+**NOTE** Make sure to add it to the top-level ``CMakeLists.txt`` file otherwise it will not be detected bu ROS
+
+.. code-block:: cmake
+
+  pluginlib_export_plugin_description_file(moveit_ros_move_group doc/examples/move_group_capabilities/plugin_description.xml)
+
+Checking the capability
+^^^^^^^^^^^^^^^^^^^^^^^
+To get a list of the all available capabilities run
+
+.. code-block:: bash
+
+  ros2 run moveit_ros_move_group list_move_group_capabilities
+
+you should see ``move_group/GetCollisionObjectsCapability`` as a capability
```

Comment:
I think it would be nice to have the expected output here. The downside is it could get out of sync if the default move group plugins change in the future.